src/main/java/com/softserveinc/dokazovi/controller/AuthController.java

```diff
@@ -84,6 +87,15 @@ public ResponseEntity<AuthResponse> authenticateUser(@Valid @RequestBody LoginRe
                     )
             );
             SecurityContextHolder.getContext().setAuthentication(authentication);
+
+            String userIp = request.getRemoteAddr();
```

Comment:
Actually, as a TLDR, on Heroku getRemoteAddr() will return IP addresses representing Heroku servers, not the end users.

Reply:
Forgot to add: make sure you can disable the X-Forwarded-For check, and use request.getRemoteAddr() as the primary source of truth. But only as an application.yaml config value!

The reason for this is when the app is NOT behind a reverse proxy, any value can be put in X-Forwarded-For instead of the real IP.